### DISCUSSION TITLE
Error when adding supabase w/ rollup: 'w3cwebsocket' is not exported by node_modules/websocket/index

### DISCUSSION THREAD

**Discussion by janzheng** [supabase/supabase#475] [Questions] [answered]
Hi there,

I'm trying to create a simple todo list with a vanilla install of Svelte/Sapper, using yarn and bundled with rollup, but I keep getting this error:

`'w3cwebsocket' is not exported by node_modules/websocket/index.js, imported by node_modules/@supabase/realtime-js/dist/module/RealtimeClient.js`

All I'm trying to do is:

```
import { createClient } from '@supabase/supabase-js'
const supabase = createClient(
  supabase_url,
  supabase_key
)
```
and I'm getting the error. I'm not even trying to use `realtime-js`, but I'm just getting the error when creating a client.

Has anyone come across this problem before? Thanks!

**Comment by janzheng** [comment, marked as answer]
Hmmmm seems to be working with the latest version of Sapper / Rollup.... Kids, sometimes don't forget to update your dependencies!

I'm getting some crazy sourcemap errors though, that I can't figure out:

```
Error when using sourcemap for reporting an error: Can't resolve original location of error.
The 'this' keyword is equivalent to 'undefined' at the top level of an ES module, and has been rewritten
3: import _createClass from "@babel/runtime/helpers/esm/createClass";
4: 
5: var __awaiter = this && this.__awaiter || function (thisArg, _arguments, P, generator) {
                   ^
6:   function adopt(value) {
7:     return value instanceof P ? value : new P(function (resolve) {
Error when using sourcemap for reporting an error: Can't resolve original location of error.
The 'this' keyword is equivalent to 'undefined' at the top level of an ES module, and has been rewritten
3: import _createClass from "@babel/runtime/helpers/esm/createClass";

...(this goes on for a long time)
```

**Reply by janzheng** [reply]
to fix this, set `context: 'this'`:
```
export default {
	client: {
		input: config.client.input(),
                context: 'this', // required for supabase realtime
```

also, sometimes removing `rollup-plugin-node-globals` and/or `rollup-plugin-node-builtins` fixes the issue

**Reply by hallee9000** [reply]
> Hmmmm seems to be working with the latest version of Sapper / Rollup.... Kids, sometimes don't forget to update your dependencies!
> 
> I'm getting some crazy sourcemap errors though, that I can't figure out:
> 
> ```
> Error when using sourcemap for reporting an error: Can't resolve original location of error.
> The 'this' keyword is equivalent to 'undefined' at the top level of an ES module, and has been rewritten
> 3: import _createClass from "@babel/runtime/helpers/esm/createClass";
> 4: 
> 5: var __awaiter = this && this.__awaiter || function (thisArg, _arguments, P, generator) {
>                    ^
> 6:   function adopt(value) {
> 7:     return value instanceof P ? value : new P(function (resolve) {
> Error when using sourcemap for reporting an error: Can't resolve original location of error.
> The 'this' keyword is equivalent to 'undefined' at the top level of an ES module, and has been rewritten
> 3: import _createClass from "@babel/runtime/helpers/esm/createClass";
> 
> ...(this goes on for a long time)
> ```

Upgrading @rollup/plugin-commonjs worked for me.
